src/vs/workbench/contrib/debug/common/debugViewModel.ts

```diff
@@ -74,6 +75,10 @@ export class ViewModel implements IViewModel {
 	setFocus(stackFrame: IStackFrame | undefined, thread: IThread | undefined, session: IDebugSession | undefined, explicit: boolean): void {
 		const shouldEmitForStackFrame = this._focusedStackFrame !== stackFrame;
 		const shouldEmitForSession = this._focusedSession !== session;
+		// currently, it does not happen that shouldEmitForThread === true, but shouldEmitForStackFrame === false.
```

Comment:
i don't understand; isn't it flipped, we select **stack frames** under **threads**?
there's two things:  when you select a thread, the model always then sets the focus to a stack frame; you can't get a 'thread focused' event currently, but i wanted to allow for it. (I didn't want to introduce that yet, because its not required for my use case, and can be added later without changing this API).

Also, during my debugging, the stack frame comparison never was 'they do equal each other' because its using _exact_ comparison, and they were always different objects. Didn't want to change due to risk, and also, the change was not needed to simply allow extensions to know about selection state.
Thanks for the comment - did this make sense? Wanted to make sure i understood your question before changing anything here.

Reply:
Yeah, I don't know what I was thinking, I looked at this for too long and got it mixed up in my head.

And re: the object comparison, I can look into it, but I would assume that object comparison should be ok, and we will for example create new stack frame objects on every step, and then it's a different frame and should fire the event.

Reply:
ok, thanks. i've removed this comment. the comment later about firing only stackframe or thread focus should be enough